libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +60,49 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	firewalldRunning, err = checkRunning()
+	if err != nil {
+		return err
+	}
+	// If firewalld is not running on the first check poll
+	if !firewalldRunning {
+		ctx, cancel := context.WithTimeout(context.Background(), fireWalldInitializationTimeout)
+		defer cancel()
+
+		for {
+
+			firewalldRunning, err = func(ctx context.Context) (bool, error) {
+
+				var running bool
+				log.G(context.TODO()).Debugf("Checking status of Firewalld")
+				running, err = checkRunning()
+
+				if err != nil {
+					return running, err
+				}
+				if ctx.Err() != nil {
+					return running, fmt.Errorf("firewalld initialization timed out")
+				}
+				return running, nil
+			}(ctx)
+
+			if firewalldRunning || err != nil {
+				break
+			}
+		}
+
+		if err != nil {
+			log.G(context.TODO()).Debugf("Check running failed: %v", err)
+		}
+	}
+
 	if !firewalldRunning {
-		connection.sysconn.Close()
-		connection = nil
```

Comment:
Why are these lines removed? We cannot configure firewalld if firewalld is not running!

Reply:
To meet a requirement Brad mentioned 

"When we hit the limit, we should act as firewalld is ready (e.g. that it left init) we log a message _and continue,_ and hope for the best"

Reply:
Here we go again...

That requirement you quoted is for the case where firewalld _is running_ (i.e. responds to the RPC call to get the `state` property) but _is not ready_ in time. I am  concerned about the other case, where firewalld _is not running_.